src/components/LoggingExportAppService/LoggingExportAppService.js

```diff
@@ -0,0 +1,167 @@
+/** Transmit queued messages when their number reaches or exceeds this amount. */
+BATCH_SIZE = 100;
+
+/** Wait at most this number of milliseconds after first message in batch is received before transmitting.*/
+TIMEOUT_MILLISECS = 30 * 1000;
+
+/** Wait at most this number of milliseconds for batch transmission to complete */
+TRANSMIT_TIMEOUT_MILLISECS = 10 * 1000;
+
+/** Log levels of messsages to capture. Valid values: Error, Warning, Log, Info, Debug, Verbose */
+CAPTURE_LOG_LEVELS = {
+	"Log": true,
+	"Warn": true,
+	"Error": true
+};
+
+/** Log categories of messages to capture. Valid values: system, dev or perf */
+CAPTURE_LOG_CATEGORIES = {
+	"system": true,
+	"dev": true
+};
+
+/**
+ * Value indicating whether log messages should be sorted by timestamp.
+ *
+ * Note that, due to the async nature of logging, you might still send a later batch containing messages with a lower
+ * logTimestamp. Hence, it is best to avoid sorting messages in this service if messages will be sorted elsewhere.
+ */
+SORT_MESSAGES = false;
+
+/**
+ * Message formatting function which converts each log message from Finsemble's format to the format you wish to transmit.
+ */
+const formatMessage = (logMessage) => {
+	/* Example Incoming log message format:
+	 *		[
+	 *			{
+	 *				"category": "system",				//Log message type: system, dev or perf
+	 *				"logClientName": "Finsemble",		//The registered name of the logger instance
+	 *				"logType": "Log",					//Log level: Error, Warning, Log, Info, Debug, Verbose
+	 *				"logData": "[\"SERVICE LIFECYCLE: STATE CHANGE: Service initializing\",\"windowService\"]",
+	 *													//JSON encoded array of message and data components of the log message
+	 *													//N.B. maybe be prefixed by string "*** Logging Error: ""
+	 *				"logTimestamp": 1544090028391.6226	//Log message timestamp for ordering use
+	 *			},
+	 *			{...},
+	 *			...
+	 *		]
+	 */
+
+	// TODO: add any necessary message format changes here
+	return logMessage;
+};
+
+
+const FSBLReady = () => {
+	try {
+		const batchAllocSize = BATCH_SIZE + 10;   //add a small amount of leeway on batch size
```

Comment:
Does all of this need to be in the `FSBLReady` function? I was thinking the `FSBLReady` function could be more like the following:
```
const FSBLReady = () => {
  try {
    // Start the consumer
		centralLoggerLogConsumer();
	} catch (e) {
		FSBL.Clients.Logger.error(e);
	}
};